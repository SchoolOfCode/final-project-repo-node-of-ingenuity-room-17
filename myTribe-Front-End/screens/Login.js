import { StatusBar } from 'expo-status-bar';
import {
	StyleSheet,
	Text,
	View,
	ImageBackground,
	TextInput,
	Button,
	KeyboardAvoidingView,
} from 'react-native';
import bgImage from '../assets/main-background.jpeg';

export default function Login() {
	return (
		<ImageBackground source={bgImage} style={styles.container}>
			{/* <Text>Open up App.js to start working on your app!</Text> */}
			<StatusBar color='white' style='light-content' />
			<View style={styles.overlay}>
				<View style={styles.header}>
					<Text style={styles.heading}>tribe.</Text>
					<Text style={styles.subHeading}>
						Your family, your rules.
					</Text>
				</View>
				<KeyboardAvoidingView behavior='padding' style={styles.authBox}>
					<TextInput
						style={styles.input}
						placeholder='youremail@email.com'
						placeholderTextColor={'black'}
					></TextInput>
					<TextInput
						style={[styles.input, styles.lastInput]}
						placeholder='password'
						placeholderTextColor={'black'}
						secureTextEntry
					></TextInput>
					<Button style={styles.btn} title='Login' color='white' />
					<Button title='Create Account' color='white' />
				</KeyboardAvoidingView>
			</View>
		</ImageBackground>
	);
}

const styles = StyleSheet.create({
	container: {
		height: '100%',
		width: '100%',
	},

	overlay: {
		flex: 1,
		justifyContent: 'space-between',
		width: '100%',
		height: '100%',
		backgroundColor: 'rgba(0,0,0,0.4)',
		color: 'white',
		paddingTop: `35%`,
		paddingLeft: 50,
		paddingRight: 50,
		paddingTop: 150,
		paddingBottom: 150,
	},

	heading: {
		fontSize: 64,
		fontWeight: 'bold',
		color: 'white',
	},
	subHeading: {
		fontSize: 24,
		color: 'white',
	},
	input: {
		width: '100%',
		height: 40,
		backgroundColor: 'white',
		borderRadius: 5,
		paddingLeft: 10,
		paddingRight: 10,
		marginTop: 15,
	},

	lastInput: {
		marginBottom: 20,
	},
});
