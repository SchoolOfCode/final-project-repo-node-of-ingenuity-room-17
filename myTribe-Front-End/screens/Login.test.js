import React from "react";
import renderer from "react-test-renderer";

import Login from "./Login";

describe("<Login />", () => {
  it("has 2 children", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree.children.length).toBe(2);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
