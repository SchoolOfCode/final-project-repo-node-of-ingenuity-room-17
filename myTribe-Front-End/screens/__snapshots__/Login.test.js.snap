// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if Login screen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.4)",
        "color": "white",
        "flex": 1,
        "height": "100%",
        "justifyContent": "space-between",
        "paddingBottom": 150,
        "paddingLeft": 50,
        "paddingRight": 50,
        "paddingTop": 150,
        "width": "100%",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 64,
            "fontWeight": "bold",
          }
        }
      >
        tribe.
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        Your family, your rules.
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "paddingBottom": 0,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        onChangeText={[Function]}
        placeholder="youremail@email.com"
        placeholderTextColor="black"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 5,
            "height": 40,
            "marginTop": 15,
            "paddingLeft": 10,
            "paddingRight": 10,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="password"
        placeholderTextColor="black"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 5,
              "height": 40,
              "marginTop": 15,
              "paddingLeft": 10,
              "paddingRight": 10,
              "width": "100%",
            },
            Object {
              "marginBottom": 20,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "color": "white",
            "marginBottom": 10,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="login"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Create Account
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
